src/endpointsApi.js

```diff
@@ -0,0 +1,149 @@
+'use strict'
+
+const logger = require('./logger')
+const EndpointModel = require('./models/endpoints')
+const {handleServerError, validateEndpoint} = require('./util')
+const KoaBodyParser = require('@viweei/koa-body-parser')
+
+const saveEndPoint = body => {
+  return new Promise((resolve, reject) => {
+    const endpoint = new EndpointModel(body)
+    endpoint.save(err => {
+      if (err) reject(err)
+      resolve(endpoint)
+    })
+  })
+}
+
+const updateEndPoint = (endpointId, body) => {
+  return new Promise((resolve, reject) => {
+    EndpointModel.findOneAndUpdate(
+      {_id: endpointId},
+      body,
+      {new: true, runValidators: true},
+      (err, result) => {
+        if (err) reject(err)
+        resolve(result)
+      }
+    )
+  })
+}
+
+const createCreateEndpoint = router => {
+  router.post('/endpoints', async (ctx, next) => {
+    const failureMsg = 'Endpoint creation/update failed: '
+
+    try {
+      await KoaBodyParser()(ctx, () => {})
+
+      if (
+        !ctx.request ||
+        !ctx.request.body ||
+        !Object.keys(ctx.request.body).length
+      ) {
+        ctx.status = 400
+        const err = `${failureMsg}Invalid endpoint object`
+        ctx.body = {error: err}
+        logger.error(err)
+        return next()
+      }
+      const body = Object.assign({}, ctx.request.body, {
+        lastUpdated: Date.now()
+      })
```

Comment:
I think this object assign can be simplified. You're creating two new objects here. The empty one isn't needed.

```suggestion
      const body = Object.assign({
        lastUpdated: Date.now()
      }, ctx.request.body)
```

Reply:
done in https://github.com/jembi/openhim-mediator-mapping/pull/48/commits/15bfc6533485b00b36bca5072290769c1973486a